my-app/src/Pages/Home.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { movieService } from "../Services/MovieService";
+import MovieList from "../Components/MovieList";
+
+const Home: React.FC<{}> = (props: any) => {
+  const queryClient = useQueryClient();
+
+  const [page, setPage] = useState(1);
+
+  const { isError, isLoading, data, isFetching, isPreviousData } = useQuery(
+    ["movies", page],
```

Comment:
don't have page in the query dependency as page will change, so you will not actually invalidate the correct query. Have ['movies'], as you only need movies here.